### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference in Overlay Driver

### ISSUE THREAD

**Issue by alvinr** [docker/libnetwork#446]
# Problem
Found the following in the logs

```
Received user event name:jl 172.16.145.140 4f8a1f6461ab2439b331eb79a5274d7351e6c0dba382f06a9109025d505d3f0b a48238eba01e77d16bc3ce6a56feb5cf1eebc4b5a026e1e2f7578318cbdeb249, payload:join 172.21.0.5 02:42:7d:af:b6:d8
Parsed data = 4f8a1f6461ab2439b331eb79a5274d7351e6c0dba382f06a9109025d505d3f0b/a48238eba01e77d16bc3ce6a56feb5cf1eebc4b5a026e1e2f7578318cbdeb249/172.16.145.140/172.21.0.5/02:42:7d:af:b6:d8
Received user event name:jl 172.16.145.143 4f8a1f6461ab2439b331eb79a5274d7351e6c0dba382f06a9109025d505d3f0b 790dfd544c462739d72e9f056595f5663e6aefd0194c0bde24ad45d54d9c4fc5, payload:join 172.21.0.4 02:42:a4:96:3c:29
Parsed data = 4f8a1f6461ab2439b331eb79a5274d7351e6c0dba382f06a9109025d505d3f0b/790dfd544c462739d72e9f056595f5663e6aefd0194c0bde24ad45d54d9c4fc5/172.16.145.143/172.21.0.4/02:42:a4:96:3c:29
2015/08/07 20:25:10 http: panic serving 172.16.145.139:38013: runtime error: invalid memory address or nil pointer dereference
goroutine 1262 [running]:
net/http.func·011()
	/usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/docker/libnetwork/drivers/overlay.(*driver).Join(0xc20823e630, 0xc20857a900, 0x40, 0xc208b9b200, 0x40, 0xc208f24780, 0x22, 0x7fa7b8a589d0, 0xc208777290, 0xc208f24750, ...)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/drivers/overlay/joinleave.go:40 +0x561
github.com/docker/libnetwork.(*endpoint).Join(0xc208777290, 0xc208741ac0, 0x40, 0xc2087dcdc0, 0x5, 0x8, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/endpoint.go:412 +0x786
github.com/docker/docker/daemon.(*Container).configureNetwork(0xc2080d52c0, 0x7ffe9da94cfb, 0x9, 0xc2089213d0, 0xa, 0x7ffe9da94cf3, 0x7, 0xc208921301, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/container_unix.go:910 +0x467
github.com/docker/docker/daemon.(*Container).AllocateNetwork(0xc2080d52c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/container_unix.go:864 +0x587
github.com/docker/docker/daemon.(*Container).initializeNetworking(0xc2080d52c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/container_unix.go:957 +0x503
github.com/docker/docker/daemon.(*Container).Start(0xc2080d52c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/container.go:269 +0x216
github.com/docker/docker/daemon.(*Daemon).ContainerStart(0xc20800cd20, 0xc2082df277, 0x40, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/start.go:43 +0x24c
github.com/docker/docker/api/server.(*Server).postContainersStart(0xc208194c00, 0xc2082df267, 0x4, 0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40, 0xc208837620, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/server/server.go:1059 +0x162
github.com/docker/docker/api/server.*Server.(github.com/docker/docker/api/server.postContainersStart)·fm(0xc2082df267, 0x4, 0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40, 0xc208837620, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/server/server.go:1651 +0x7b
github.com/docker/docker/api/server.func·008(0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40)
	/go/src/github.com/docker/docker/api/server/server.go:1598 +0xc8f
net/http.HandlerFunc.ServeHTTP(0xc2080a9a40, 0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40)
	/usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc20818aaf0, 0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40)
	/go/src/github.com/docker/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc2081823c0, 0x7fa7b8a5dd98, 0xc208492f00, 0xc2084a3d40)
	/usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2082f5d60)
	/usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e
```

# Workaround
None (yet)

# Environment

```
$ uname -a
Linux swarm-2 4.0.3-boot2docker #1 SMP Wed May 13 20:54:49 UTC 2015 x86_64 GNU/Linux
```

```
$ docker version
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   5fdc102
 Built:        Wed Jul 22 23:59:08 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   5fdc102
 Built:        Wed Jul 22 23:59:08 UTC 2015
 OS/Arch:      linux/amd64
 Experimental: true
```

```
$ docker info
Containers: 2
Images: 52
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 56
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.3-boot2docker
Operating System: Boot2Docker 1.6.2 (TCL 5.4); master : 4534e65 - Wed May 13 21:24:28 UTC 2015
CPUs: 1
Total Memory: 998 MiB
Name: swarm-2
ID: 5ZT3:SV33:WJVM:OMQW:RFX7:K3Q5:MQ5H:CMRL:A6ZP:OT7J:HDCI:32WA
Debug mode (server): true
File Descriptors: 34
Goroutines: 83
System Time: 2015-08-07T20:30:49.539665532Z
EventsListeners: 1
Init SHA1: 
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=vmwarefusion
 com.docker.network.driver.overlay.neighbor_ip=172.16.145.139
 com.docker.network.driver.overlay.bind_interface=eth0
Experimental: true
```

**Comment by aboch** [comment]
Fixed by #454

**Comment by mavenugo** [comment]
@alvinr can you pls confirm ?
